sources/advanced/options/utils.ts

```diff
@@ -24,13 +24,16 @@ export type Tuple<Type, Arity extends number> = Arity extends Arity
     : TupleOf<Type, Arity, []>
   : never;
 
-export type WithArity<Type, Arity extends number> = Arity extends 0
-  ? boolean
-  : Arity extends 1
-    ? Type
-    : number extends Arity
-      ? boolean | Type | Tuple<Type, Arity>
-      : Tuple<Type, Arity>;
+export type WithArity<Type extends {length?: number}, Arity extends number> =
+  number extends Type['length']
```

Comment:
I'm not 100% it's the perfect approach (basically if the type is already a tuple it's returned as-is rather than being repeated N times), but it should be a good approximation until a bug is found, if ever.